### PR TITLE
Enable/Disable Endpoint timestamps

### PR DESCRIPTION
**Pull request by MatthewBerk**
**What**:
Added timestamp to response message when enable/disable chaos monkey with the endpoints `/chaosmonkey/enable` and `/chaosmonkey/disable`.
Modified test cases for `enableChaosMonkey()` and  ` disableChaosMonkey()` methods in order to handle timestamp now being present.

**Why**:
https://github.com/codecentric/chaos-monkey-spring-boot/issues/246#issuecomment-885985299

**How**:
Using java.time.Clock to get current time of user and attaching it to the already present "Chaos Monkey is enabled/disabled" message.

**Checklist**:

- [ ] Documentation added `N/A`
      
- [x] Changelog updated
      
- [ ] Tests added `N/A`
      
- [x] Ready to be merged

### CODE REVIEW

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyJmxEndpoint.java
Comment (WtfJoke) [reply chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyJmxEndpoint.java:83]: Now that I see it, I would preferr the term "enabledAt" over "activatedAt" as we already use the verb enabled in "Chaos Monkey is enabled".


I think it would make sense to use StringJoiner, it would make the code more readable:
```java
new StringJoiner("\n")
            .add("Chaos Monkey is enabled")
            .add("enabledAt: " + ZonedDateTime.now())
            .toString();
 ````
 
However I think even better would be to introduce a new class (POJO, eg. ChaosMonkeyEnabledDto and ChaosMonkeyDisabledDto) and return that instead. Then we dont need to deal with "\n" and formats  and spring/jackson can take care of the conversion of that object to a string :)
 
 Then it could look like this:
 ```java
   @WriteOperation
  public ChaosMonkeyEnabledDto enableChaosMonkey() {
    this.chaosMonkeySettings.getChaosMonkeyProperties().setEnabled(true);
    return new ChaosMonkeyEnabledDto();
  }
 ```
 
 ```java
 @Data
public class ChaosMonkeyEnabledDto {
    private final String status = "Chaos Monkey is enabled";
    private ZonedDateTime enabledAt = ZonedDateTime.now();
}
```

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeRestEndpointIntegration.java
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeRestEndpointIntegration.java:278]: As soon as we introduced an object (see comment above). You could make assertions again. 
it would be great if we make a basic assertion that the correct time would be returned.

We could for example remember the time before we send the request and make an assertion that the time is returned should be later than the remembered time.

**Review by WtfJoke — COMMENTED** [review]
Hey @MatthewBerk 

I left feedback in the comments. Left some ideas for improving the code. 
Let me know if something is unclear

File: chaos-monkey-docs/src/main/asciidoc/changes.adoc
Comment (MatthewBerk) [reply chaos-monkey-docs/src/main/asciidoc/changes.adoc:7]: Will do. Had pushed the day before that change got merged!

File: chaos-monkey-docs/src/main/asciidoc/changes.adoc
Comment (MatthewBerk) [reply chaos-monkey-docs/src/main/asciidoc/changes.adoc:15]: I was debating between the two! The reason I went with new feature since I was adding a new element to Chaos Monkey. Will make the changes!

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyJmxEndpoint.java
Comment (MatthewBerk) [reply chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyJmxEndpoint.java:83]: The class's is a good idea. I recall seeing ZonedDateTime.now(). Will need to check again to figure out why didn't go with it!

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeRestEndpointIntegration.java
Comment (MatthewBerk) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeRestEndpointIntegration.java:278]: So it would still be assertTrue but just checking that the time stored in object returned is later than time recorded before assertion statement?

Comment (WtfJoke) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeRestEndpointIntegration.java:278]: Yeah it could be two asserts. One which asserts the text/status "Chaos Monkey is enabled" and one the time yes

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java
Comment (MatthewBerk) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java:118]: The reason I used isAfter and isEqual is because every time I ran the test, except one time, it had isAfter false and isEqual true! So just in case others encounter that odd event, I didn't want them to get a failed test!

**Review by WtfJoke — COMMENTED** [review]
Looks pretty good so far. 👍   Have only some smaller remarks on the Tests.  After that this can get merged :)

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java
Comment (MatthewBerk) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java:110]: It says method doesn't exist when did a quick check!

Comment (WtfJoke) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java:110]: oh my bad. The method name is a bit different. Its `isAfterOrEqualTo`

Comment (MatthewBerk) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java:110]: Cannot resolve method 'assertThat(java.time.ZonedDateTime)'

Comment (WtfJoke) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java:110]: I see why, we usually use assertj, but in this test hamcrest is still used. 
Assertj has this method, hamcrest not. 

I thought we got rid of hamcrest completly, but it seems a few tests still uses it.

Comment (WtfJoke) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java:110]: You can keep it that way it is or change it to assertj, I'll leave it up to you.

I most likely would do a PR which removes hamcrest completly (as it's causes such confusions like here :D)

Comment (MatthewBerk) [reply chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/endpoints/ChaosMonkeyRequestScopeJmxEndpointTest.java]: I updated it! Though I left  `import static org.hamcrest.MatcherAssert.assertThat;` since the other tests needed it!

**Review by WtfJoke — APPROVED** [review]
LGTM 🕐 

Thank you @MatthewBerk

**Pull request merged**
